### PR TITLE
actually rename courses csp and csd with -2017 suffixes

### PR DESCRIPTION
**Pull request by davidsbailey**
### Description
* executes step 2 outlined in [dev doc](https://docs.google.com/document/d/1Os6zB01TophoitiOdE6quf5SH1H0TW4gZFdv4EknMrg/edit#)

### Timing considerations
* do not merge until https://github.com/code-dot-org/code-dot-org/pull/22511 has been deployed, so that we can seamlessly keep serving the CSP 2017 course during deployment. If these end up in the same DTP, then it's possible we'd serve 404s at /courses/csp after seeding happens and before the code change kicks in.
* It is safe for the code change and course rename within this PR to be deployed together. The code change is seeding logic which will be executed on production-daemon, not production code that needs to reach frontends before taking effect.

### CODE REVIEW

**Comment by davidsbailey** [comment]
It looks like tests are having problems because `ScriptConstants.full_courses` is busted -- it's assumed to contain course versions (e.g. csp-2018) in some cases, and course families (e.g. csp) in other cases, and only works right now because 'csp' is currently both a course family and a course version. I think I need to fix this in a different PR which will be merged before this one, so that we can address anything that breaks due to that change separately.

**Comment by davidsbailey** [comment]
Apologies for the push spam. PTAL @balderdash 

Since this was reviewed last, this PR has been rebased on top of:
* https://github.com/code-dot-org/code-dot-org/pull/22511 redirect from csp to csp-2017
* https://github.com/code-dot-org/code-dot-org/pull/22576 fix full courses

The first 6 commits are mostly the same as previously reviewed, and the last 7 commits are new. Of the new commits, only https://github.com/code-dot-org/code-dot-org/pull/22519/commits/1be81af659fcbca8a12c668d606ade9d1fc9e8c5 is substantial, and the assignment selector was broken without it. This functionality is mostly covered by [teacher_homepage.feature](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/features/teacher_homepage.feature) .

@ewjordan please take a quick peek at contact_rollups.rb !

File: lib/cdo/contact_rollups.rb [hunk 1]
```diff
@@ -463,8 +463,8 @@ def self.update_roles
     add_role_from_script_sections_taught("CSF Teacher", CSF_SCRIPT_LIST)
     # CSD and CSP scripts are mapped to course - identify CSD/CSP teachers
     # that way
-    add_role_from_course_sections_taught("CSD Teacher", "csd")
-    add_role_from_course_sections_taught("CSP Teacher", "csp")
+    add_role_from_course_sections_taught("CSD Teacher", "csd-2017")
+    add_role_from_course_sections_taught("CSP Teacher", "csp-2017")
```
Comment (ewjordan) [inline lib/cdo/contact_rollups.rb:467]: (the change to this file) LGTM, I didn't see any other areas in `contact_rollups` that directly referenced these!

File: pegasus/helpers/section_api_helpers.rb [hunk 2]
```diff
@@ -393,7 +393,7 @@ def self.valid_courses
       select(:id, :name).
       all.
       # Only return courses we've whitelisted in ScriptConstants
-      select {|course| ScriptConstants.script_in_category?(:full_course, course_assignment_family(course))}.
+      select {|course| ScriptConstants.script_in_category?(:full_course, course[:name])}.
```
Comment (balderdash) [inline pegasus/helpers/section_api_helpers.rb:396]: Unrelated to this change, but it's unfortunate that we're passing a course name into a method called `script_in_category?`.

Comment (davidsbailey) [reply pegasus/helpers/section_api_helpers.rb:396]: Agreed 😞 It's also sad that this is in module called ScriptConstants. Our term for "course or script" is "assignment", so perhaps we could start using that term here in the future.

**Comment by davidsbailey** [comment]
🤞

**Pull request merged**